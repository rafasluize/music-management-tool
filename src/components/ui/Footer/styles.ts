import styled from 'styled-components';

const Wrapper = styled.header`
  padding: ${({ theme }) => theme.spacing.km4};

  text-align: center;
  background-color: ${({ theme }) => theme.color.background.neutral.strong};
`;

export { Wrapper };
