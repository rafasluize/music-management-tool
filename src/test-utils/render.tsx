import React, { FC, JSXElementConstructor, ReactElement, ReactNode, Suspense } from 'react';
import { ThemeProvider } from 'styled-components';
import { render, RenderOptions } from '@testing-library/react';
import theme from 'styles/theme';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { BrowserRouter } from 'react-router-dom';

const ThemeDS: FC<{ children: ReactNode }> = ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>;

export const wrapperRender = (
  component: ReactElement<Element, string | JSXElementConstructor<Element>>,
  options?: RenderOptions<typeof import('@testing-library/dom/types/queries'), HTMLElement>,
) => render(component, { wrapper: ThemeDS, ...options });

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

const queryRender = (
  component: ReactElement<Element, string | JSXElementConstructor<Element>>,
  defaultValues?: unknown,
  options?: RenderOptions<typeof import('@testing-library/dom/types/queries'), HTMLElement>,
) =>
  render(
    <Suspense fallback={<p>Carregando</p>}>
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>{component}</BrowserRouter>
      </QueryClientProvider>
    </Suspense>,
    { wrapper: ThemeDS, ...options },
  );

export { wrapperRender as render, queryRender };

export default wrapperRender;
