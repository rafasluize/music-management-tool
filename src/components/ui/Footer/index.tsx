import React, { FC } from 'react';

import * as Styled from './styles';

const Footer: FC = () => {
  const date = new Date().getFullYear();
  return <Styled.Wrapper>{date} - All Rights Reserved</Styled.Wrapper>;
};

export default Footer;
