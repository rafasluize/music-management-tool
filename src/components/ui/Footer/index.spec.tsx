import React from 'react';
import { render } from 'test-utils/render';
import Footer from '.';

describe('footer', () => {
  it('snapshot', () => {
    const { container } = render(<Footer />);

    // eslint-disable-next-line testing-library/no-node-access
    const wrapper = container.firstElementChild;

    expect(wrapper).toMatchSnapshot();
  });
});
