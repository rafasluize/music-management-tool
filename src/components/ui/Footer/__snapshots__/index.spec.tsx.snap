// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer snapshot 1`] = `
.c0 {
  padding: 16px;
  text-align: center;
  background-color: #dddfe4;
}

<header
  class="c0"
>
  2024
   - All Rights Reserved
</header>
`;
